pkg/deviceclaimingserver/enddevices/ttjsv2/ttjs.go

```diff
@@ -108,10 +108,11 @@ var (
 	errBadRequest           = errors.DefineInvalidArgument("bad_request", "bad request", "message")
 	errDeviceNotProvisioned = errors.DefineNotFound("device_not_provisioned", "device with EUI `{dev_eui}` not provisioned") //nolint:lll
 	errDeviceNotClaimed     = errors.DefineNotFound("device_not_claimed", "device with EUI `{dev_eui}` not claimed")
-	errDeviceAccessDenied   = errors.DefinePermissionDenied("device_access_denied", "access to device with `{dev_eui}` denied: device is already claimed or the owner token is invalid") //nolint:lll
-	errUnauthenticated      = errors.DefineUnauthenticated("unauthenticated", "unauthenticated")
+	errDeviceAccessDenied   = errors.DefineInvalidArgument("device_access_denied", "access to device with `{dev_eui}` denied: device is already claimed or the owner token is invalid") //nolint:lll
+	errUnauthenticated      = errors.DefineInternal("unauthenticated", "unauthenticated")
```

Comment:
I think this is a mistake. This should be reverted since we have a dedicated internal error `errInternalError`.

Reply:
So if our client is unauthenticated, it is an internal configuration error.

Reply:
Should I use `errInternalError` instead of `errUnauthenticated` to make it clear that will produce an "internal error" with error code 500  or should I leave it like this to produce "unauthenticated" error with error code 500?

Reply:
I think it is okay like this. We return an internal error, but the reason is that TTS is not authenticated with TTJS. So we see the right error in the logs and we see the right error in the response.

Reply:
This isn't great for readability but this is fine as it is.

Reply:
@halimi okay can you rename this to `errCredentials` instead?

Reply:
Sure, done.